File: android/src/main/java/io/openim/flutter_openim_sdk/listener/AdvancedMsgListener.java
package io.openim.flutter_openim_sdk.listener;

import android.util.ArrayMap;

import java.util.Map;

import io.openim.flutter_openim_sdk.util.CommonUtil;
import open_im_sdk.OnAdvancedMsgListener;

public class AdvancedMsgListener implements OnAdvancedMsgListener {
    //    final Map<String, String> values = new HashMap<>();
    private final String id;

    public AdvancedMsgListener(String listenerId) {
        this.id = listenerId;
//        this.values.put("id", listenerId);
    }

    @Override
    public void onRecvC2CReadReceipt(String s) {
        final Map<String, String> values = new ArrayMap<>();
        values.put("id", id);
        values.put("haveReadMessage", s);
        CommonUtil.emitEvent("advancedMsgListener", "onRecvC2CReadReceipt", values);
    }

    @Override
    public void onRecvMessageRevoked(String s) {
        final Map<String, String> values = new ArrayMap<>();
        values.put("id", id);
        values.put("revokedMessage", s);
        CommonUtil.emitEvent("advancedMsgListener", "onRecvMessageRevoked", values);
    }

    @Override
    public void onRecvNewMessage(String s) {
        final Map<String, String> values = new ArrayMap<>();
        values.put("id", id);
        values.put("newMessage", s);
        CommonUtil.emitEvent("advancedMsgListener", "onRecvNewMessage", values);
    }
}

File: lib/src/listener/impl/advanced_msg_listener.dart
import 'package:flutter_openim_sdk/flutter_openim_sdk.dart';

class OnAdvancedMsgListener extends AdvancedMsgListener {
  Function(List<HaveReadInfo> list)? onRecvC2CReadReceipt;
  Function(String msgId)? onRecvMessageRevoked;
  Function(Message msg)? onRecvNewMessage;

  OnAdvancedMsgListener({
    this.onRecvC2CReadReceipt,
    this.onRecvMessageRevoked,
    this.onRecvNewMessage,
  });

  @override
  void recvC2CReadReceipt(List<HaveReadInfo> list) {
    if (null != onRecvC2CReadReceipt) onRecvC2CReadReceipt!(list);
  }

  @override
  void recvMessageRevoked(String msgId) {
    if (null != onRecvMessageRevoked) onRecvMessageRevoked!(msgId);
  }

  @override
  void recvNewMessage(Message msg) {
    if (null != onRecvNewMessage) onRecvNewMessage!(msg);
  }
}
